strawberry_django/relay.py

```diff
@@ -75,18 +97,59 @@ def resolve_connection(
         last: Optional[int] = None,
         **kwargs: Any,
     ) -> AwaitableOrValue[Self]:
-        # FIXME: Asynchronous queryset iteration is only available on Django 4.1+.
-        # Remove this when Django we can remove support for django 4.0 and older
-        if django.VERSION < (4, 1) and in_async_context():
-            return sync_to_async(cls.resolve_connection)(  # type: ignore
-                nodes,
-                info=info,
-                before=before,
-                after=after,
-                first=first,
-                last=last,
-                **kwargs,
-            )
+        if (
+            isinstance(nodes, models.QuerySet)
+            and (result := nodes._result_cache) is not None  # type: ignore
+        ):
```

Comment:
not really sure what's happening here, could we also move this code to a new method?